Segment-Intercom/Classes/SEGIntercomIntegration.m

```diff
@@ -74,10 +74,48 @@ - (void)track:(SEGTrackPayload *)payload
         SEGLog(@"[Intercom logEventWithName:%@];", payload.event);
         return;
     }
-    [self.intercom logEventWithName:payload.event metaData:payload.properties];
-    SEGLog(@"[Intercom logEventWithName:%@ metaData:%@];", payload.event, payload.properties);
+
+
+    NSMutableDictionary *original = [NSMutableDictionary dictionaryWithDictionary:payload.properties];
+    NSMutableDictionary *output = [NSMutableDictionary dictionaryWithCapacity:payload.properties.count];
+
+    NSMutableDictionary *price = [NSMutableDictionary dictionaryWithCapacity:0];
+
+    [original enumerateKeysAndObjectsUsingBlock:^(id key, id data, BOOL *stop) {
+
+        [output setObject:data forKey:key];
+        if ([key isEqual:@"revenue"] || [key isEqual:@"total"]) {
+            double dataValue = [data doubleValue];
+            long amountInCents = dataValue * 100;
```

Comment:
Why long rather than integer here? Not sure how the data types are different in iOS. In Java, integers count up to 2,147,483,647, which should prob be sufficient for a revenue value.